p4include/v1model.p4

```diff
@@ -151,8 +181,12 @@ extern void truncate(in bit<32> length);
 parser Parser<H, M>(packet_in b,
                     out H parsedHdr,
                     inout M meta,
-                    inout standard_metadata_t standard_metadata);
-control VerifyChecksum<H, M>(in H hdr,
+inout standard_metadata_t standard_metadata);
+
+/* The only legal statements in the implementation of the
+VerifyChecksum control are: block statements, calls to the
+verify_checksum method, and return statements. */
+control VerifyChecksum<H, M>(inout H hdr,
```

Comment:
Why did this first parameter become `inout`?  Is it legal for `VerifyChecksum` to change headers?

Reply:
There was an explanation in the original PR. This is because in BMv2 you cannot use a constant for the checksum value. It *must* be a (p4-14) field. So you cannot, for example, check that the checksum is 0.

So, while we do not allow the block to change the checksum, this is how we enforce it being a left-value.

Reply:
Alternatively, we could keep the parameter "in", but make another separate check in the back-end for the argument to the verify_checksum function.

Reply:
Couldn't you create a field temporary in the backend if the parameter is a constant?

Reply:
Not a big deal, but logically it seems that the result of verifying a checksum is a boolean -- either the checksum is valid or it is not.  I guess in this case you've added a single field to match the checksum against and in most uses it will always be a constant 0 or ~0?  This allows for weird non-standard checksum where the result is something else, but that bmv2 does not handle as well.

Reply:
These constraints come all from modelling the way checksums are defined in p4-14 (and in consequence in bmv2), where the checksum has to indicate a fieldlist, a field, and then whether the checksum is verified or updated.